src/Composer/ComposerWidget.vala

```diff
@@ -358,6 +352,36 @@ public class Mail.ComposerWidget : Gtk.Grid {
 
             to_grid_style_context.set_state (state);
         });
+
+        if (to != null) {
+            to_val.text = to;
+        }
+
+        if (mailto_query != null) {
+            var result = new Gee.HashMap<string, string> ();
+            var params = mailto_query.split ("&");
+
+            foreach (unowned string param in params) {
+                var terms = param.split ("=");
+                if (terms.length != 2) {
+                    critical ("Invalid mailto URL");
+                }
+
+                result[terms[0]] = Soup.URI.decode (terms[1]);
```

Comment:
This should be in an `else` clause, I think. I think it might even be a crash if `terms.length < 2`?

Reply:
Nice catch!